buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -332,6 +332,17 @@ class PerformanceTestPlugin : Plugin<Project> {
             teamCityPassword = stringPropertyOrNull(PropertyNames.teamCityPassword)
         }
 
+        afterEvaluate {
+            result.configure {
+                branchName?.takeIf { it.isNotEmpty() }?.let { branchName ->
+                    channel = channel + "-" + branchName
```

Comment:
I think I like overriding `getChannel` in `DistributedPerformanceTest` better than re-setting it in an after evaluate block. WDYT?

Reply:
Yes, though you need to move the logic for setting the system property to the task execute method in that case. Or turn the whole thing into an ArgumentProvider

Reply:
Ah, I didn't see the `setChannel` method also provides a system property to the test VM. I guess `setBranch` would then also need to update the system property?

Anyway, let's leave it in `afterEvaluate`.

Reply:
My understanding is this would be part of the default value on Provider stream of work. Basically, a way to bind some default logic to be taken into account. I wrongly assumed the `afterEvaluate` block was going to get executed if a project is evaluated, but it's actually nested `afterEvaluate` block that has been recently fixed.

With a Provider, you could have the logic moved in there and get rid of the `afterEvaluate` block as it would be evaluated when the provider is realized which is only if the task is actually going to execute.

Reply:
I doesn't seem to be a default value for the provider, since `channel` needs also to be set explicitly. The actual "channel" provider should probably be a different property, which is composed by the `channel` property and the `branch` property. And this new "channel" provide would provide a system property.